app/questionnaire/router.py

```diff
@@ -267,7 +288,7 @@ def _get_last_complete_section_key(self) -> Tuple[str, Optional[str]]:
             if self._progress_store.is_section_complete(section_id, list_item_id):
                 return section_id, list_item_id
 
-    def _is_section_enabled(self, section):
+    def _is_section_enabled(self, section: Mapping) -> Union[bool, date, None]:
```

Comment:
`date`?. This should always return a bool?.

Reply:
This one depends on `WhenRuleEvaluator` methods which don't return `bool` hence these types in router. I'll try to fix'em in evaluator.

Reply:
Ya this is a consequence of the evaluator supporting boolean and value operators, but I think we should make code changes so that the typing is explicit otherwise we are just making it harder for ourselves. We know it should only be a `bool` if the schema is valid.

So 2 options, either check if is an instance of a bool, and raise exception if not, or cast to a bool which will always give true or false (it shouldn't ever be not a bool) but this will help us have strict typing. Casting is easier and doesn't require any tests or ignore line coverage.

Reply:
I've added casting to bool in `_is_section_enabled` as other approach would would mean refactoring old `when rule evaluator` and don't we want to get rid of it soon anyway? Thanks!